src/annotations/api/index.ts

```diff
@@ -22,7 +22,19 @@ import {formatAnnotationQueryString} from 'src/annotations/utils/formatQueryStri
 export const writeAnnotation = async (
   annotations: Annotation[]
 ): Promise<Annotation[]> => {
-  const res = await axios.post(url, annotations)
+  // we need to convert the annotation object's start and end time fields to be
+  // string types so they can be parsed in the backend.
+  const annotationsRequestConverted = annotations.map(annotation => {
```

Comment:
so we'll want to remove these ignores - we can get TypeScript to shut up in other ways that are a little simpler.

the problem here, is that `annotations` is an `Annotation[]` (an `Annotation` array) and `Annotations` have dates as numbers, but we're trying to take an array of `Annotations` and change their number dates to strings. The problem occurs because you're trying to re-assign `annotation.start`.

Overwriting the arguments that come in from a function is generally an antipattern; you should copy them instead:

```js
const annotationsRequestConverted = annotations.map(annotation => {
  return {
    ...annotation,
    start: new Date(annotation.start).toISOString(),
    end: new Date(annotation.end).toISOString(),
  }
})
```

```js
const original = { name: 'Sahas' }
const poser = { ...original }

poser.name = 'Bucky'
console.log(original, poser)
// { name: "Sahas" }  { name: "Bucky" }
```

Reply:
Thank you for this solid detailed explanation Bucky! I now understand what the 3 dots do! I read up on it as well and it's very interesting how it can be used to write cleaner code.